src/throttler/throttler.ts

```diff
@@ -31,6 +31,12 @@ export interface ThrottlerStats {
   elapsed: number;
 }
 
+interface TaskData<T> {
+  task: T;
+  retryCount: number;
+  wait: { resolve: (result: any) => void; reject: (err: Error) => void } | undefined;
```

Comment:
```typescript
wait?: { resolve: (result: any) => void; reject: (err: Error) => void };
```

optional parameters can be defined with `?` rather than `| undefined`